src/include/daos_prop.h

```diff
@@ -648,6 +648,19 @@ daos_prop_free(daos_prop_t *prop);
 int
 daos_prop_from_str(const char *str, daos_size_t len, daos_prop_t **prop);
 
+/**
+ * Merge a set of new DAOS properties into a set of existing DAOS properties.
+ *
+ * \param[in]	old_prop	Existing set of properties
+ * \param[in]	new_prop	New properties - may override old entries
+ * \param[out]	out_prop	New properties - may override old entries
+ *
+ * \return		0		Success
+ *			-DER_NOMEM	Out of memory
+ */
+int
+daos_prop_merge2(daos_prop_t *old_prop, daos_prop_t *new_prop, daos_prop_t **out_prop);
```

Comment:
so we are adding a new API just to support this fault injection framework?
i agree generally this new API looks better; but if that is the only cause it sounds unnecessary to start versioning in the API just because of that.

Reply:
i don't think this API was used anywhere other than internally. i wonder if we can just get away by breaking the API here. probably not a good suggestion :-)

Reply:
I think what I've done is a good way forward, this is not a major issue.  Maintaining an API over multiple versions can be a pain.

Reply:
And it should be unaffected by this PR.